### DISCUSSION TITLE
FoxyTabs Compatibility

### DISCUSSION THREAD

**Discussion by EduCryptor** [mbnuqw/sidebery#1560] [General]
## Description
Is it possible to have Sideberry function in conjunction with FoxyTabs? I'm aware that the problem is that Firefox doesn't recognize the Sideberry "tabs" as regular tabs. I can deduce that's the reason Sideberry can use it's own context menu, which is very intuitive, by the way. 

But is there a way, at the very least in theory, to edit the recognition patterns (the way that it is recognized by FoxyTabs, or what FoxyTabs recognizes to begin with)? What is the ContextType of the Sideberry "tabs/links/objects/whatever they are?"

Check out to following issue to better understand my request:
https://github.com/erosman/support/issues/589

**Comment by albino1** [comment]
Reading over your issue, it seems like just using the native context menu with Sidebery would solve your problem?

`Settings` -> `Context Menu` -> `Use native context menu`

If you want both, then just keep using Sidebery's built-in context menu, and then `Shift+Right-Click` to get the native one whenever you need it.

**Comment by EduCryptor** [comment]
> Reading over your issue, it seems like just using the native context menu with Sidebery would solve your problem?
> 
> `Settings` -> `Context Menu` -> `Use native context menu`
> 
> If you want both, then just keep using Sidebery's built-in context menu, and then `Shift+Right-Click` to get the native one whenever you need it.

Thank you for replying! If you read over the issue I referenced in the link on the original post, you'll see that I mention my awareness of this "Combination-Click" capability. However, the problem is not that I can't access the native context menu, it is that FoxyTabs does not recognize the Sideberry "tabs" as native tabs (because apparently Sideberry "tabs" are NOT native tabs, nor are they the same ContextType as the native tabs).

_Tree Style Tabs_ allows FoxyTabs to recognize the tabs on the sidebar because I presume that the ContextType of the _TST_ tabs are the same as Firefox's native tabs. But I don't ever want to switch to anything. I want to keep using Sideberry, since it's just the best of its kind. See (https://github.com/erosman/support/issues/575) for more details about the _TST_ compatibility.

**Comment by albino1** [comment]
Ah, I see.  Sorry that didn't help, hopefully something can be done.

**Comment by mbnuqw** [comment]
Sidebery already uses the "tab" ContextType for native context menu (https://github.com/mbnuqw/sidebery/blob/358bcaf0218c624e22d4eac4d7aba9e6efd1cc86/src/sidebar/components/tab.vue#L330). If you want to use options of other addons you just need to switch to the native context menu as @albino1 suggested (Sidebery settings / Context menu / Use native context menu).

Using context menu options of other addons in non-native context menu of sidebery is impossible.

**Comment by EduCryptor** [comment]
> Sidebery already uses the "tab" ContextType for native context menu (
> 
> https://github.com/mbnuqw/sidebery/blob/358bcaf0218c624e22d4eac4d7aba9e6efd1cc86/src/sidebar/components/tab.vue#L330
> ). If you want to use options of other addons you just need to switch to the native context menu as @albino1 suggested (Sidebery settings / Context menu / Use native context menu).
> 
> Using context menu options of other addons in non-native context menu of sidebery is impossible.

> Reading over your issue, it seems like just using the native context menu with Sidebery would solve your problem?
> 
> `Settings` -> `Context Menu` -> `Use native context menu`
> 
> If you want both, then just keep using Sidebery's built-in context menu, and then `Shift+Right-Click` to get the native one whenever you need it.

This was actually the perfect solution for me, except for one thing:
All of my add-ons appear in the ContextMenu when I 'Shift+Right-Click', EXCEPT for FoxyTabs. I was lead to believe that it was an issue with Sideberry, but I had my doubts from the beginning that it was Sideberry's fault. This is the best add-on ever created!

Any idea how I can make FoxyTabs appear in the ContextMenu when I `Shift+Right-Click`?

https://github.com/mbnuqw/sidebery/assets/126362441/930f1a03-164e-496e-86ba-9a68b6946586

**Comment by mbnuqw** [comment]
You don't need to `Shift+RightClick`. Just open sidebery settings and activate native context menu.

https://github.com/mbnuqw/sidebery/assets/6276694/f9f67832-11a3-4688-b9fb-16d6e9b7d65f

**Comment by EduCryptor** [comment]
> You don't need to `Shift+RightClick`. Just open sidebery settings and activate native context menu.
> Peek.2023-08-31.21-29.mp4

Thank you! Of this I'm aware; I was able to do it on my own. I've configured everything on my Sidebery to be a certain way, including the context menu. I switched it, but I prefer Sidebery's context menu so I switched it back. The problem is that I want to be able to access the FoxyTabs options using `Shift+Right-Click`, but I'm not able to. I understand the workaround, but why does the specific problem in question exist, and why are other add-ons showing their menu options when I `Shift+Right-Click`?

**Comment by mbnuqw** [comment]
Ok, now I understand the issue. It's impossible for addon to handle and configure context-menu event fired with `Shift+Right-Click`, which is a direct purpose of `Shift+Right-Click` - forcefully show context menu for the page. So, when you `Shift+Right-Click` on sidebery (doesn't matter what area) browser think you right-clicked on the page and shows you a context menu with options of addons that work on normal web-pages.

**Comment by EduCryptor** [comment]
> Ok, now I understand the issue. It's impossible for addon to handle and configure context-menu event fired with `Shift+Right-Click`, which is a direct purpose of `Shift+Right-Click` - forcefully show context menu for the page. So, when you `Shift+Right-Click` on sidebery (doesn't matter what area) browser think you right-clicked on the page and shows you a context menu with options of addons that work on normal web-pages.

That is an excellent explanation! That makes total sense. However, would you happen to understand why Firefox's "Multi-Account Containers" context menu options pop up when it's an add-on that works specifically for tabs? In other words, how come I can see the containers add-on menu options and not the FoxyTabs menu options when they're both concerned with tab functionality, albeit in slightly different ways? You can see it pop up in the video I posted earlier.

**Comment by mbnuqw** [comment]
"Multi-Account Containers" (MAC) shows context menu option because when you're reopening a site in some container with MAC, it creates context-menu options for the `all` context-type if tab of the target container is active. So wherever you right-click on the page in container (or sidebar with `shift`), you will get this context menu option. (maybe there are other cases)

And FoxyTab doesn't create a context menu option for the `all` context-type, only for the `tab`.

**Comment by EduCryptor** [comment]
https://github.com/erosman/support/issues/589#issuecomment-1703908001

How does this ^^ square into the picture?
